events/src/camera.rs

```diff
@@ -31,10 +35,8 @@ pub struct Camera {
     /// Physical interface: `usb`, `csi`, `ethernet`
     pub interface: Option<String>,
 
-    /// URI to access video:
-    /// - local file name for `local` cameras (for example `/dev/video0`)
-    /// - RTSP URI for `remote` cameras (for example `rtsp://192.168.1.2/my_stream`)
-    pub uri: Option<String>,
+    /// RTSP stream address (`remote` cameras only).
+    pub rtsp_uri: Option<Url>,
```

Comment:
This seems like a hack to have rtsp_uri even if the camera will not have RTSP URI. Why not have separate types for diff types of cameras and have a trait for common API? The `uri` accessor methods in the trait should treat it as optional and the USB camera type can keep it mandatory.

Reply:
It is just the first PR of the refactoring, I think there'll be enum with all camera types.

Reply:
cool. Sorry for being pedantic but how does this change lead to/help with the goal of separate types?

Reply:
Hm, let me think. This change is not about splitting `Camera` into separate types, it's about not loosing RTSP URI in cases when it's needed. There's a bit of [code](https://github.com/lumeohq/lumeod/pull/39/files#diff-c8d72bf2092bec87ffa983a761c12507R54) in lumeod that is used for filling this exact field, just don't want to touch it for now (aside from renaming `uri` to `rtsp_uri`).

Reply:
> This change is not about splitting Camera into separate types

Yeah, I understand. What I was saying was that this seems very much like a hack that becomes redundant with the proper fix of splitting the type. So unless there is some urgency here, I'd rather we already implement the proper fix instead.

Reply:
Well, let's see how it looks when I split types then. So, I close this PR.

Reply:
Thanks and sorry for being annoying. :)